src/macros.rs

```diff
@@ -1,3 +1,37 @@
+macro_rules! define_uuid_macro {
+    {$(#[$doc:meta])*} => {
+        $(#[$doc])*
+        #[cfg(feature = "macros")]
+        #[macro_export]
+        macro_rules! uuid {
+            ($uuid:literal) => {{
+                $crate::Uuid::from_bytes($crate::uuid_macro::parse_lit!($uuid))
+            }};
+        }
+
+        $(#[$doc])*
+        #[cfg(not(feature = "macros"))]
+        #[macro_export]
+        macro_rules! uuid {
+            ($uuid:literal) => {{
+                const OUTPUT: $crate::Uuid = match $crate::Uuid::try_parse($uuid) {
+                    Ok(u) => u,
+                    Err(_) => {
+                        // here triggers const_err
+                        // const_panic requires 1.57
+                        #[allow(unconditional_panic)]
+                        let _ = ["invalid uuid representation"][1]; 
+
+                        loop {} // -> never type
```

Comment:
😂

Reply:
Tricky, but it works :joy:
<https://docs.rs/static_assertions/1.1.0/src/static_assertions/const_assert.rs.html#52-57>